mbed_targets/mbed_targets.py

```diff
@@ -59,3 +59,21 @@ def __iter__(self) -> Iterator["MbedTarget"]:
         """Yield an MbedTarget on each iteration."""
         for target in self._target_data:
             yield MbedTarget(target)
+
+    def get_target(self, product_code: str) -> "MbedTarget":
+        """Look up an MbedTarget by its product code.
+
+        Args:
+            product_code: the product code.
+
+        Raises:
+            ValueError: the given product code was not found in the target database.
+        """
+        try:
+            return next(
+                target for target in self if target.product_code == product_code
+            )
+        except StopIteration:
+            raise ValueError(
```

Comment:
Would this be better as a tools exception, perhaps `UnknownTarget`? I could imagine this could be a common exception.